<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>STAT 9610 Lecture Notes - 25&nbsp; Poisson regression</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./negative-binomial-regression.html" rel="next">
<link href="./logistic-regression.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>

  <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
      <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./glm-special-cases.html">Generalized linear models: Special cases</a></li><li class="breadcrumb-item"><a href="./poisson-regression.html"><span class="chapter-number">25</span>&nbsp; <span class="chapter-title">Poisson regression</span></a></li></ol></nav>
      <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
      </a>
      <button type="button" class="btn quarto-search-button" aria-label="" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">STAT 9610 Lecture Notes</a> 
        <div class="sidebar-tools-main">
    <a href="./STAT-9610-Lecture-Notes.pdf" rel="" title="Download PDF" class="quarto-navigation-tool px-1" aria-label="Download PDF"><i class="bi bi-file-pdf"></i></a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Introduction</span></span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="./linear-models-estimation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Linear models: Estimation</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./interpreting-linear-models.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Interpreting linear models</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./least-squares-estimation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Least squares estimation</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./anova.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Analysis of variance</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./collinearity.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Collinearity and adjustment</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./r-demo-part-1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">R demo</span></span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="./linear-models-inference.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Linear models: Inference</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./building-blocks.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Building blocks</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./hypothesis-testing.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Hypothesis testing</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./power.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">9</span>&nbsp; <span class="chapter-title">Power</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./confidence-intervals.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">10</span>&nbsp; <span class="chapter-title">Confidence intervals</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./practical-considerations.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">11</span>&nbsp; <span class="chapter-title">Practical considerations</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./r-demo-part-2.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">12</span>&nbsp; <span class="chapter-title">R demo</span></span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="./linear-models-misspecification.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Linear models: Misspecification</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./misspecification-overview.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">13</span>&nbsp; <span class="chapter-title">Overview</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./asymptotic-methods.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">14</span>&nbsp; <span class="chapter-title">Asymptotic methods</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./bootstrap.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">15</span>&nbsp; <span class="chapter-title">The bootstrap</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./permutation-test.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">16</span>&nbsp; <span class="chapter-title">The permutation test</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./robust-estimation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">17</span>&nbsp; <span class="chapter-title">Robust estimation and inference</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./r-demo-part-3.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">18</span>&nbsp; <span class="chapter-title">R demo</span></span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="./glm-general-theory.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Generalized linear models: General theory</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./exponential-dispersion-models.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">19</span>&nbsp; <span class="chapter-title">Exponential dispersion model (EDM) distributions</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./generalized-linear-models.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">20</span>&nbsp; <span class="chapter-title">GLM definition</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./parameter-estimation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">21</span>&nbsp; <span class="chapter-title">Parameter estimation</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./glm-inference.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">22</span>&nbsp; <span class="chapter-title">Inference in GLMs</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./r-demo-part-4.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">23</span>&nbsp; <span class="chapter-title">R demo</span></span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="./glm-special-cases.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Generalized linear models: Special cases</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-5" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./logistic-regression.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">24</span>&nbsp; <span class="chapter-title">Logistic regression</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./poisson-regression.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">25</span>&nbsp; <span class="chapter-title">Poisson regression</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./negative-binomial-regression.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">26</span>&nbsp; <span class="chapter-title">Negative binomial regression</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./r-demo-part-5.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">27</span>&nbsp; <span class="chapter-title">R demo</span></span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="./multiple-testing.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Multiple testing</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-6" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-6" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./multiple-testing-intro.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">28</span>&nbsp; <span class="chapter-title">Introduction</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./global-testing.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">29</span>&nbsp; <span class="chapter-title">Global testing</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./multiple-testing-chapter.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">30</span>&nbsp; <span class="chapter-title">Multiple testing</span></span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#sec-model-definition" id="toc-sec-model-definition" class="nav-link active" data-scroll-target="#sec-model-definition"><span class="header-section-number">25.1</span> Model definition and interpretation</a></li>
  <li><a href="#sec-example-modeling-rates" id="toc-sec-example-modeling-rates" class="nav-link" data-scroll-target="#sec-example-modeling-rates"><span class="header-section-number">25.2</span> Example: Modeling rates</a></li>
  <li><a href="#sec-estimation-inference" id="toc-sec-estimation-inference" class="nav-link" data-scroll-target="#sec-estimation-inference"><span class="header-section-number">25.3</span> Estimation and inference</a>
  <ul class="collapse">
  <li><a href="#sec-wald-inference" id="toc-sec-wald-inference" class="nav-link" data-scroll-target="#sec-wald-inference"><span class="header-section-number">25.3.1</span> Score, Fisher information, and Wald inference</a></li>
  <li><a href="#sec-likelihood-ratio-inference" id="toc-sec-likelihood-ratio-inference" class="nav-link" data-scroll-target="#sec-likelihood-ratio-inference"><span class="header-section-number">25.3.2</span> Likelihood ratio inference</a></li>
  <li><a href="#sec-score-based-inference" id="toc-sec-score-based-inference" class="nav-link" data-scroll-target="#sec-score-based-inference"><span class="header-section-number">25.3.3</span> Score-based inference</a></li>
  </ul></li>
  <li><a href="#sec-poisson-multinomial" id="toc-sec-poisson-multinomial" class="nav-link" data-scroll-target="#sec-poisson-multinomial"><span class="header-section-number">25.4</span> Relationship between Poisson and multinomial distributions</a></li>
  <li><a href="#sec-example-2x2-tables" id="toc-sec-example-2x2-tables" class="nav-link" data-scroll-target="#sec-example-2x2-tables"><span class="header-section-number">25.5</span> Example: <span class="math inline">\(2 \times 2\)</span> contingency tables</a>
  <ul class="collapse">
  <li><a href="#sec-poisson-2x2" id="toc-sec-poisson-2x2" class="nav-link" data-scroll-target="#sec-poisson-2x2"><span class="header-section-number">25.5.1</span> Poisson model for <span class="math inline">\(2 \times 2\)</span> contingency tables</a></li>
  <li><a href="#sec-inference-conditioning-margins" id="toc-sec-inference-conditioning-margins" class="nav-link" data-scroll-target="#sec-inference-conditioning-margins"><span class="header-section-number">25.5.2</span> Inference is the same regardless of conditioning on margins</a></li>
  <li><a href="#sec-poisson-logistic-equivalence" id="toc-sec-poisson-logistic-equivalence" class="nav-link" data-scroll-target="#sec-poisson-logistic-equivalence"><span class="header-section-number">25.5.3</span> Equivalence among Poisson and logistic regressions</a></li>
  </ul></li>
  <li><a href="#sec-poisson-jk-tables" id="toc-sec-poisson-jk-tables" class="nav-link" data-scroll-target="#sec-poisson-jk-tables"><span class="header-section-number">25.6</span> Example: Poisson models for <span class="math inline">\(J \times K\)</span> contingency tables</a></li>
  <li><a href="#sec-poisson-jkl-tables" id="toc-sec-poisson-jkl-tables" class="nav-link" data-scroll-target="#sec-poisson-jkl-tables"><span class="header-section-number">25.7</span> Example: Poisson models for <span class="math inline">\(J \times K \times L\)</span> contingency tables</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title"><span id="sec-poisson-regression" class="quarto-section-identifier"><span class="chapter-number">25</span>&nbsp; <span class="chapter-title">Poisson regression</span></span></h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<section id="sec-model-definition" class="level2" data-number="25.1">
<h2 data-number="25.1" class="anchored" data-anchor-id="sec-model-definition"><span class="header-section-number">25.1</span> Model definition and interpretation</h2>
<p>The Poisson regression model (with offsets) is:</p>
<p><span id="eq-poisson-with-offsets"><span class="math display">\[
y_i \overset{\text{ind}}\sim \text{Poi}(\mu_i); \quad \log \mu_i = o_i + \boldsymbol{x}_{i*}^T \boldsymbol{\beta}.
\tag{25.1}\]</span></span></p>
<p>Because the log of the mean is linear in the predictors, Poisson regression models are often called <em>loglinear models</em>. To interpret the coefficients, note that a unit increase in <span class="math inline">\(x_j\)</span> (while keeping the other variables fixed) is associated with an increase in the predicted mean by a factor of <span class="math inline">\(e^{\beta_j}\)</span>.</p>
</section>
<section id="sec-example-modeling-rates" class="level2" data-number="25.2">
<h2 data-number="25.2" class="anchored" data-anchor-id="sec-example-modeling-rates"><span class="header-section-number">25.2</span> Example: Modeling rates</h2>
<p>One cool feature of the Poisson model is that rates can be easily modeled with the help of offsets. Let’s say that the count <span class="math inline">\(y_i\)</span> is collected over the course of a time interval of length <span class="math inline">\(t_i\)</span>, or a spatial region with area <span class="math inline">\(t_i\)</span>, or a population of size <span class="math inline">\(t_i\)</span>, etc. Then, it is meaningful to model:</p>
<p><span class="math display">\[
y_i \overset{\text{ind}} \sim \text{Poi}(\pi_i t_i), \quad \log \pi_i = \boldsymbol{x}^T_{i*}\boldsymbol{\beta},
\]</span></p>
<p>where <span class="math inline">\(\pi_i\)</span> represents the rate of events per day / per square mile / per capita, etc. In other words:</p>
<p><span class="math display">\[
y_i \overset{\text{ind}} \sim \text{Poi}(\mu_i), \quad \log \mu_i = \log t_i + \boldsymbol{x}^T_{i*}\boldsymbol{\beta},
\]</span></p>
<p>which is exactly equation (<a href="#eq-poisson-with-offsets"><span>25.1</span></a>) with offsets <span class="math inline">\(o_i = \log t_i\)</span>. For example, in single-cell RNA-sequencing, <span class="math inline">\(y_i\)</span> is the number of RNA molecules aligning to a gene in cell <span class="math inline">\(i\)</span> and <span class="math inline">\(t_i\)</span> is the total number of RNA molecules measured in the cell, a quantity called the <em>library size</em>. We might use a Poisson regression model to carry out a <em>differential expression analysis</em> between two cell types.</p>
</section>
<section id="sec-estimation-inference" class="level2" data-number="25.3">
<h2 data-number="25.3" class="anchored" data-anchor-id="sec-estimation-inference"><span class="header-section-number">25.3</span> Estimation and inference</h2>
<section id="sec-wald-inference" class="level3" data-number="25.3.1">
<h3 data-number="25.3.1" class="anchored" data-anchor-id="sec-wald-inference"><span class="header-section-number">25.3.1</span> Score, Fisher information, and Wald inference</h3>
<p>We found in Chapter 4 that the score and Fisher information for Poisson regression are:</p>
<p><span class="math display">\[
\boldsymbol{U}(\boldsymbol{ \beta}) = \boldsymbol{X}^T(\boldsymbol{y} - \boldsymbol{\mu}),
\]</span></p>
<p>and:</p>
<p><span class="math display">\[
\boldsymbol{I}(\boldsymbol{\beta}) = \boldsymbol{X}^T \boldsymbol{W} \boldsymbol{X} = \boldsymbol{X}^T \text{diag}(V(\mu_i))\boldsymbol{X} = \boldsymbol{X}^T \text{diag}(\mu_i)\boldsymbol{X},
\]</span></p>
<p>respectively. Hence, the normal equations for the MLE are:</p>
<p><span class="math display">\[
\boldsymbol{X}^T(\boldsymbol{y} - \boldsymbol{\hat \mu}).
\]</span></p>
<p>Wald inference is based on the approximation:</p>
<p><span class="math display">\[
\boldsymbol{\hat \beta} \overset \cdot \sim N(\boldsymbol{\beta}, (\boldsymbol{X}^T \text{diag}(\hat \mu_i)\boldsymbol{X})^{-1}).
\]</span></p>
</section>
<section id="sec-likelihood-ratio-inference" class="level3" data-number="25.3.2">
<h3 data-number="25.3.2" class="anchored" data-anchor-id="sec-likelihood-ratio-inference"><span class="header-section-number">25.3.2</span> Likelihood ratio inference</h3>
<p>For likelihood ratio inference, we first derive the total deviance. The unit deviance of a Poisson distribution is:</p>
<p><span class="math display">\[
d(y, \mu) = y \log \frac{y}{\mu} - (y - \mu).
\]</span></p>
<p>Hence, the total deviance is:</p>
<p><span class="math display">\[
D(\boldsymbol{y}, \boldsymbol{\mu}) = \sum_{i = 1}^n d(y_i, \mu_i) = \sum_{i = 1}^n \left(y_i \log \frac{y_i}{\mu_i} - (y_i - \mu_i)\right).
\]</span></p>
<p>The residual deviance is then:</p>
<p><span class="math display">\[
D(\boldsymbol{y}, \boldsymbol{\hat\mu}) = \sum_{i = 1}^n \left(y_i \log \frac{y_i}{\hat \mu_i} - (y_i - \hat \mu_i)\right) = \sum_{i = 1}^n y_i \log \frac{y_i}{\hat \mu_i}.
\]</span></p>
<p>The last equality holds for any model containing the intercept, since by the normal equations we have <span class="math inline">\(\sum_{i = 1}^n (y_i - \hat \mu_i) = \boldsymbol{1}^T (\boldsymbol{y} - \boldsymbol{\hat \mu}) = 0\)</span>. We can carry out a likelihood ratio test for <span class="math inline">\(H_0: \boldsymbol \beta_S = \boldsymbol{0}\)</span> via:</p>
<p><span class="math display">\[
D(\boldsymbol{y}, \boldsymbol{\hat \mu}^0) - D(\boldsymbol{y}, \boldsymbol{\hat \mu}) = \sum_{i = 1}^n y_i \log \frac{\hat \mu_i}{\hat \mu^0_{i}} \overset{\cdot}\sim \chi^2_{|S|}.
\]</span></p>
<p>We can carry out a goodness-of-fit test via:</p>
<p><span class="math display">\[
D(\boldsymbol{y}, \boldsymbol{\hat\mu}) = \sum_{i = 1}^n y_i \log \frac{y_i}{\hat \mu_i} \overset{\cdot}\sim \chi^2_{n - p}.
\]</span></p>
</section>
<section id="sec-score-based-inference" class="level3" data-number="25.3.3">
<h3 data-number="25.3.3" class="anchored" data-anchor-id="sec-score-based-inference"><span class="header-section-number">25.3.3</span> Score-based inference</h3>
<p>Recalling <a href="glm-inference.html#sec-score-test-single-coeff"><span>Section&nbsp;22.4.2</span></a>, the score test for <span class="math inline">\(H_0: \beta_j = 0\)</span> is based on the approximation <span class="math display">\[
\frac{\boldsymbol{x}_{*,j}^T (\boldsymbol{y} - \boldsymbol{\widehat{\mu}}^0)}{\sqrt{\boldsymbol x_{*,j}^T \boldsymbol {\widehat W}^{0}\boldsymbol x_{*,j} - \boldsymbol x_{*,j}^T \boldsymbol {\widehat W}^{0}\boldsymbol X_{*,\text{-}j}(\boldsymbol X_{*,\text{-}j}^T \boldsymbol {\widehat W}^{0}\boldsymbol X_{*,\text{-}j})^{-1}\boldsymbol X_{*,\text{-}j}^T \boldsymbol {\widehat W}^{0}\boldsymbol x_{*,j}}} \overset{\cdot}\sim N(0,1),
\]</span> where <span class="math display">\[
\boldsymbol {\widehat W}^{0} = \text{diag}(\boldsymbol{\widehat{\mu}}^0).
\]</span></p>
<p>On the other hand, the score test for goodness-of-fit is based on the Pearson <span class="math inline">\(X^2\)</span> statistic:</p>
<p><span class="math display">\[
X^2 \equiv \sum_{i = 1}^n \frac{(y_i - \hat \mu_i)^2}{\hat \mu_i} \overset{\cdot}\sim \chi^2_{n - p}.
\]</span></p>
</section>
</section>
<section id="sec-poisson-multinomial" class="level2" data-number="25.4">
<h2 data-number="25.4" class="anchored" data-anchor-id="sec-poisson-multinomial"><span class="header-section-number">25.4</span> Relationship between Poisson and multinomial distributions</h2>
<p>Suppose that <span class="math inline">\(y_i \overset{\text{ind}}\sim \text{Poi}(\mu_i)\)</span> for <span class="math inline">\(i = 1, \dots, n\)</span>. Then,</p>
<p><span class="math display">\[
\begin{split}
\mathbb{P}\left[y_1 = m_1, \dots, y_n = m_n \left| \sum_{i}y_i = m\right.\right] &amp;= \frac{\mathbb{P}[y_1 = m_1, \dots, y_n = m_n]}{\mathbb{P}[\sum_{i}y_i = m]} \\
&amp;= \frac{\prod_{i = 1}^n e^{-\mu_i}\frac{\mu_i^{m_i}}{m_i!}}{e^{-\sum_i \mu_i}\frac{(\sum_i \mu_i)^m}{m!}} \\
&amp;= {m \choose m_1, \dots, m_n} \prod_{i = 1}^n \pi_i^{m_i},
\end{split}
\]</span> where <span class="math display">\[
\pi_i \equiv \frac{\mu_i}{\sum_{i' = 1}^n \mu_{i'}}.
\]</span> We recognize the last expression in the previous display as the probability mass function of the multinomial distribution with parameters <span class="math inline">\((\pi_1, \dots, \pi_n)\)</span> summing to one. In words, the joint distribution of a set of independent Poisson distributions conditional on their sum is a multinomial distribution.</p>
</section>
<section id="sec-example-2x2-tables" class="level2" data-number="25.5">
<h2 data-number="25.5" class="anchored" data-anchor-id="sec-example-2x2-tables"><span class="header-section-number">25.5</span> Example: <span class="math inline">\(2 \times 2\)</span> contingency tables</h2>
<section id="sec-poisson-2x2" class="level3" data-number="25.5.1">
<h3 data-number="25.5.1" class="anchored" data-anchor-id="sec-poisson-2x2"><span class="header-section-number">25.5.1</span> Poisson model for <span class="math inline">\(2 \times 2\)</span> contingency tables</h3>
<p>Let’s say that we have two binary random variables <span class="math inline">\(x_1, x_2 \in \{0,1\}\)</span> with joint distribution <span class="math inline">\(\mathbb{P}(x_1 = j, x_2 = k) = \pi_{jk}\)</span> for <span class="math inline">\(j,k \in \{0,1\}\)</span>. We collect a total of <span class="math inline">\(n\)</span> samples from this joint distribution and summarize the counts in a <span class="math inline">\(2 \times 2\)</span> table, where <span class="math inline">\(y_{jk}\)</span> is the number of times we observed <span class="math inline">\((x_1, x_2) = (j,k)\)</span>, so that:</p>
<p><span class="math display">\[
(y_{00}, y_{01}, y_{10}, y_{11})|n \sim \text{Mult}(n, (\pi_{00}, \pi_{01}, \pi_{10}, \pi_{11})).
\]</span></p>
<p>Our primary question is whether these two random variables are independent, i.e.</p>
<p><span id="eq-null-product-formulation"><span class="math display">\[
\pi_{jk} = \pi_{j+}\pi_{+k},
\tag{25.2}\]</span></span> where <span class="math display">\[
\pi_{j+} \equiv \mathbb{P}[x_1 = j] = \pi_{j1} + \pi_{j2}; \quad \pi_{+k} \equiv \mathbb{P}[x_2 = k] = \pi_{1k} + \pi_{2k}.
\]</span></p>
<p>We can express this equivalently as:</p>
<p><span class="math display">\[
\pi_{00}\pi_{11} = \pi_{01}\pi_{10}.
\]</span></p>
<p>In other words, we can express the independence hypothesis concisely as:</p>
<p><span id="eq-independence-2"><span class="math display">\[
H_0: \frac{\pi_{11}\pi_{00}}{\pi_{10}\pi_{01}} = 1.
\tag{25.3}\]</span></span></p>
<p>Let’s arbitrarily assume that, additionally, <span class="math inline">\(n \sim \text{Poi}(\mu_{++})\)</span>. Then, by the relationship between Poisson and multinomial distributions, we have:</p>
<p><span class="math display">\[
y_{jk} \overset{\text{ind}} \sim \text{Poi}(\mu_{++}\pi_{jk}).
\]</span></p>
<p>Let <span class="math inline">\(i \in \{1,2,3,4\}\)</span> index the four pairs <span class="math display">\[
(x_1, x_2) \in \{(0,0), (0,1), (1,0), (1,1)\},
\]</span> so that for <span class="math inline">\(i = 1, \dots, 4\)</span>, we have <span id="eq-2x2-poisson-reg"><span class="math display">\[
y_i \overset{\text{ind}}\sim \text{Poi}(\mu_i); \quad \log \mu_i = \beta_0 + \beta_1 x_{i1} + \beta_2 x_{i2} + \beta_{12}x_{i1} x_{i2},
\tag{25.4}\]</span></span></p>
<p>where:</p>
<p><span id="eq-2x2-poisson-reg-coefs"><span class="math display">\[
\begin{aligned}
\beta_0 = \log \mu_{++} + \log \pi_{00}; \quad \beta_1 = \log \frac{\pi_{10}}{\pi_{00}}; \\
\beta_2 = \log \frac{\pi_{01}}{\pi_{00}}; \quad \beta_{12} = \log\frac{\pi_{11}\pi_{00}}{\pi_{10}\pi_{01}}.
\end{aligned}
\tag{25.5}\]</span></span></p>
<p>Note that the independence hypothesis (<a href="#eq-independence-2"><span>25.3</span></a>) reduces to the hypothesis <span class="math inline">\(H_0: \beta_{12} = 0\)</span>:</p>
<p><span class="math display">\[
H_0: \frac{\pi_{11}\pi_{00}}{\pi_{10}\pi_{01}} = 1 \quad \Longleftrightarrow \quad H_0: \beta_{12} = 0.
\]</span></p>
<p>So the presence of an interaction in the Poisson regression is equivalent to a lack of independence between <span class="math inline">\(x_1\)</span> and <span class="math inline">\(x_2\)</span>. We can test the latter hypothesis using our standard tools for Poisson regression.</p>
<p>For example, we can use the Pearson <span class="math inline">\(X^2\)</span> goodness-of-fit test. To apply this test, we must find the fitted means under the null hypothesis model:</p>
<p><span id="eq-2x2-poisson-reg-null"><span class="math display">\[
y_i \overset{\text{ind}}\sim \text{Poi}(\mu_i); \quad \log \mu_i = \beta_0 + \beta_1 x_{i1} + \beta_2 x_{i2}, \quad i = 1, \dots, 4.
\tag{25.6}\]</span></span></p>
<p>The normal equations give us the following:</p>
<p><span class="math display">\[
\begin{aligned}
y_{++} &amp;\equiv \sum_{j, k = 0}^1 y_{jk} = \sum_{j, k = 0}^1 \hat \mu_{jk} \equiv \hat \mu_{++}; \\
y_{+1} &amp;\equiv \sum_{j = 0}^1 y_{j1} = \sum_{j = 0}^1 \hat \mu_{j1} \equiv \hat \mu_{+1}; \\
y_{1+} &amp;\equiv \sum_{k = 0}^1 y_{1k} = \sum_{k = 0}^1 \hat \mu_{1k} \equiv \hat \mu_{1+}.
\end{aligned}
\]</span></p>
<p>By combining these equations, we arrive at:</p>
<p><span class="math display">\[
\hat \mu_{++} = y_{++}; \quad \hat \mu_{j+} = y_{j+} \text{ for all } j \in \{0, 1\}; \quad \hat \mu_{+k} = y_{+k} \text{ for all } k \in \{0, 1\}.
\]</span></p>
<p>Therefore, the fitted means under the null hypothesis model <a href="#eq-2x2-poisson-reg-null"><span>25.6</span></a> are:</p>
<p><span class="math display">\[
\hat \mu_{jk} = \hat \mu_{++}\hat \pi_{jk} = \hat \mu_{++}\hat \pi_{j+}\hat \pi_{+k} = y_{++}\frac{y_{j+}}{y_{++}}\frac{y_{+k}}{y_{++}} = \frac{y_{j+}y_{+k}}{y_{++}}.
\]</span></p>
<p>Hence, we have:</p>
<p><span class="math display">\[
X^2 = \sum_{j,k = 0}^1 \frac{(y_{jk} - \widehat \mu_{jk})^2}{\widehat \mu_{jk}} = \sum_{j, k = 0}^1 \frac{(y_{jk} - y_{j+}y_{+k}/y_{++})^2}{y_{j+}y_{+k}/y_{++}}.
\]</span></p>
<p>Alternatively, we can use the likelihood ratio test, which gives:</p>
<p><span class="math display">\[
G^2 = 2\sum_{j,k = 0}^1 y_{jk}\log\frac{y_{jk}}{\widehat \mu_{jk}} = 2\sum_{j, k = 0}^1 y_{jk}\log\frac{y_{jk}}{y_{j+}y_{+k}/y_{++}}.
\]</span></p>
<p>We would compare both <span class="math inline">\(X^2\)</span> and <span class="math inline">\(G^2\)</span> to a <span class="math inline">\(\chi^2_1\)</span> distribution.</p>
</section>
<section id="sec-inference-conditioning-margins" class="level3" data-number="25.5.2">
<h3 data-number="25.5.2" class="anchored" data-anchor-id="sec-inference-conditioning-margins"><span class="header-section-number">25.5.2</span> Inference is the same regardless of conditioning on margins</h3>
<p>Now, our data might actually have been collected such that <span class="math inline">\(n \sim \text{Poi}(\mu_{++})\)</span>, or maybe <span class="math inline">\(n\)</span> was fixed in advance. Is the Poisson inference proposed above actually valid in the latter case? In fact, it is! To see this, let us consider the log likelihoods of the two models:</p>
<p><span class="math display">\[
p_{\boldsymbol{\mu}}(\boldsymbol{y}) = p_{\mu_{++}}(y_{++} = n)p_{\boldsymbol{\pi}}(\boldsymbol{y} | y_{++} = n),
\]</span></p>
<p>so:</p>
<p><span class="math display">\[
\begin{split}
\log p_{\boldsymbol{\mu}}(\boldsymbol{y}) &amp;= \log p_{\mu_{++}}(y_{++} = n) + \log p_{\boldsymbol{\pi}}(\boldsymbol{y} | y_{++} = n) \\
&amp;= C + \log p_{\boldsymbol{\pi}}(\boldsymbol{y} | y_{++} = n).
\end{split}
\]</span></p>
<p>In other words, the log-likelihoods of the Poisson and multinomial models, as a function of <span class="math inline">\(\boldsymbol{\pi}\)</span>, differ from each other by a constant. Therefore, any likelihood-based inference in these models is equivalent. The same argument shows that conditioning on the row or column totals (as opposed to the overall total) also yields the same exact inference. Therefore, regardless of the sampling mechanism, we can always conduct an independence test in a <span class="math inline">\(2 \times 2\)</span> table via a Poisson regression.</p>
</section>
<section id="sec-poisson-logistic-equivalence" class="level3" data-number="25.5.3">
<h3 data-number="25.5.3" class="anchored" data-anchor-id="sec-poisson-logistic-equivalence"><span class="header-section-number">25.5.3</span> Equivalence among Poisson and logistic regressions</h3>
<p>We’ve talked about two ways to view a <span class="math inline">\(2 \times 2\)</span> contingency table. In the logistic regression view, we thought about one variable as a predictor and the other as a response, seeking to test whether the predictor has an impact on the response. In the Poisson regression view, we thought about the two variables symmetrically, seeking to test independence. It turns out that these two perspectives are equivalent. Recall that we have derived in equations <a href="#eq-2x2-poisson-reg"><span>25.4</span></a> and <a href="#eq-2x2-poisson-reg-coefs"><span>25.5</span></a> that <span class="math inline">\(x_1 \perp \!\!\! \perp x_2\)</span> if and only if <span class="math inline">\(\beta_{12} = 0\)</span> in the Poisson regression:</p>
<p><span class="math display">\[
\log y_i \overset{\text{ind}}{\sim} \text{Poi}(\mu_i), \quad \log \mu_i = \beta_0 + \beta_1 x_{i1} + \beta_2 x_{i2} + \beta_{12} x_{i1}x_{i2}, \quad i = 1, \dots, 4.
\]</span></p>
<p>However, we have:</p>
<p><span class="math display">\[
\begin{split}
\beta_{12} &amp;= \log \frac{\pi_{11}\pi_{00}}{\pi_{01}\pi_{10}} \\
&amp;= \log \frac{\pi_{11}/\pi_{01}}{\pi_{01}/\pi_{00}} = \log \frac{\mathbb{P}[x_2 = 1 \mid x_1 = 1] / \mathbb{P}[x_2 = 0 \mid x_1 = 1]}{\mathbb{P}[x_2 = 1 \mid x_1 = 0] / \mathbb{P}[x_2 = 0 \mid x_1 = 0]}.
\end{split}
\]</span></p>
<p>Recalling the logistic regression of <span class="math inline">\(x_2\)</span> on <span class="math inline">\(x_1\)</span>:</p>
<p><span id="eq-2x2-logistic-reg"><span class="math display">\[
\text{logit } \mathbb{P}[x_2 = 1 \mid x_1] = \tilde{\beta_0} + \tilde{\beta_1} x_1,
\tag{25.7}\]</span></span></p>
<p>and that <span class="math inline">\(\tilde{\beta_1}\)</span> is the log odds ratio, we conclude that:</p>
<p><span class="math display">\[
\beta_{12} = \tilde{\beta_1},
\]</span></p>
<p>so <span class="math inline">\(x_1 \perp \!\!\! \perp x_2\)</span> if and only if <span class="math inline">\(\tilde{\beta_1} = 0\)</span>. Due to the equivalence between Poisson and multinomial distributions, the hypothesis tests and confidence intervals for the log odds ratio <span class="math inline">\(\beta_{12}\)</span> (or <span class="math inline">\(\tilde{\beta_1}\)</span>) obtained from Poisson and logistic regressions will be the same.</p>
</section>
</section>
<section id="sec-poisson-jk-tables" class="level2" data-number="25.6">
<h2 data-number="25.6" class="anchored" data-anchor-id="sec-poisson-jk-tables"><span class="header-section-number">25.6</span> Example: Poisson models for <span class="math inline">\(J \times K\)</span> contingency tables</h2>
<p>Suppose now that <span class="math inline">\(x_1 \in \{1, \dots, J\}\)</span> and <span class="math inline">\(x_2 \in \{1, \dots, K\}\)</span>. Then, we denote <span class="math inline">\(\mathbb{P}[x_1 = j, x_2 = k] = \pi_{jk}\)</span>. We still are interested in testing for independence between <span class="math inline">\(j\)</span> and <span class="math inline">\(k\)</span>, which amounts to a goodness-of-fit test for the Poisson model:</p>
<p><span class="math display">\[
y_{jk} \overset{\text{ind}}\sim \text{Poi}(\mu_{jk}); \quad \log \mu_{jk} = \beta_0 + \beta^1_j + \beta^2_k.
\]</span></p>
<p>The score (Pearson) and deviance-based goodness-of-fit statistics for this test are:</p>
<p><span class="math display">\[
X^2 = \sum_{j = 1}^J \sum_{k = 1}^K \frac{(y_{jk} - \widehat \mu_{jk})^2}{\widehat \mu_{jk}} \quad \text{and} \quad G^2 = 2\sum_{j = 1}^J \sum_{k = 1}^K y_{jk}\log \frac{y_{jk}}{\widehat \mu_{jk}},
\]</span></p>
<p>where <span class="math inline">\(\widehat \mu_{jk} = \widehat y_{++}\frac{y_{j+}}{y_{++}}\frac{y_{+k}}{y_{++}}\)</span>. Like with the <span class="math inline">\(2 \times 2\)</span> case, the test is the same regardless of whether we condition on the row sums, column sums, total count, or if we do not condition at all. The degrees of freedom in the full model is <span class="math inline">\(JK\)</span>, while the degrees of freedom in the partial model is <span class="math inline">\(J + K - 1\)</span>, so the degrees of freedom for the goodness-of-fit test is <span class="math inline">\(JK - J - K + 1 = (J - 1)(K - 1)\)</span>. Pearson erroneously concluded that the test had <span class="math inline">\(JK - 1\)</span> degrees of freedom, which, when Fisher corrected it, created a lot of animosity between these two statisticians.</p>
</section>
<section id="sec-poisson-jkl-tables" class="level2" data-number="25.7">
<h2 data-number="25.7" class="anchored" data-anchor-id="sec-poisson-jkl-tables"><span class="header-section-number">25.7</span> Example: Poisson models for <span class="math inline">\(J \times K \times L\)</span> contingency tables</h2>
<p>These ideas can be extended to multi-way tables, for example, three-way tables. If we have <span class="math inline">\(x_1 \in \{1, \dots, J\}\)</span>, <span class="math inline">\(x_2 \in \{1, \dots, K\}\)</span>, <span class="math inline">\(x_3 \in \{1, \dots, L\}\)</span>, then we might be interested in testing several kinds of null hypotheses:</p>
<ul>
<li>Mutual independence: <span class="math inline">\(H_0: x_1 \perp \!\!\! \perp x_2 \perp \!\!\! \perp x_3\)</span>.</li>
<li>Joint independence: <span class="math inline">\(H_0: x_1 \perp \!\!\! \perp (x_2, x_3)\)</span>.</li>
<li>Conditional independence: <span class="math inline">\(H_0: x_1 \perp \!\!\! \perp x_2 \mid x_3\)</span>.</li>
</ul>
<p>These three null hypotheses can be shown to be equivalent to the Poisson regression model:</p>
<p><span class="math display">\[
y_{jkl} \overset{\text{ind}}\sim \text{Poi}(\mu_{jkl}),
\]</span></p>
<p>where:</p>
<p><span class="math display">\[
\log \mu_{jkl} = \beta_0 + \beta^1_j + \beta^2_k + \beta^3_l \quad \text{(mutual independence)};
\]</span></p>
<p><span class="math display">\[
\log \mu_{jkl} = \beta_0 + \beta^1_j + \beta^2_k + \beta^3_l + \beta^{2,3}_{kl} \quad \text{(joint independence)};
\]</span></p>
<p><span class="math display">\[
\log \mu_{jkl} = \beta_0 + \beta^1_j + \beta^2_k + \beta^3_l + \beta^{1,3}_{jl} + \beta^{2,3}_l \quad \text{(conditional independence)}.
\]</span></p>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="./logistic-regression.html" class="pagination-link">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">24</span>&nbsp; <span class="chapter-title">Logistic regression</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./negative-binomial-regression.html" class="pagination-link">
        <span class="nav-page-text"><span class="chapter-number">26</span>&nbsp; <span class="chapter-title">Negative binomial regression</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->



</body></html>